### ISSUE TITLE
camera failed in some devices

### ISSUE THREAD

**Issue by dat0106** [livekit/client-sdk-android#179]
**Describe the bug**
camera in some devices (Redmi note 11pro, Redmi note 11, one plus) does not show in other device,

**Expected behavior**

**Screenshots**
![20230209_163503 (1)](https://user-images.githubusercontent.com/4971539/217777527-cbcf771b-5984-4f3c-8052-d1ee06bf2b87.jpg)

**Device Info:**
 - Device: Redmi note 11pro
 - OS:  Android 12
 - LiveKit SDK version: 1.1.6

**Comment by Test-isom** [comment]
I'am experiencing same issue on redmi note 7 device and changing version of protobuf  to "3.21.12" solves it for me, can you please release a new version by merging this fix if possible @davidliu

**Comment by Test-isom** [comment]
Yeah i was going through his comment and even saw another videos he uploaded on youtube for issue, but for now i added these lines in build.gradle along with the livekit sdk dependency
and since grade would take latest version of dependency if there are conflicting version so it works for me for now on redmi note devices.

//For https://github.com/livekit/client-sdk-android/issues/179
implementation "com.google.protobuf:protobuf-javalite:3.21.12"

But thanks a lot for the reply, will be actively following the repo for a better fix,until than this works for me so i guess no need for a updated sdk.

**Comment by davidliu** [comment]
Protobuf updated to 3.22.0 in #186.

**Comment by lihongyang166** [comment]
@MikuMikuQAQ   Thank you very much, it has been solved！

**Comment by davidliu** [comment]
Hmm, looks like there's some issues here, perhaps with either the simulcast encoding or software encoding. I'll dig into this a little more and see if we can fix it so that the overrides aren't needed.

**Comment by davidliu** [comment]
Looks like this device has issues with H264 simulcast, and then switches to VP8 encoding on the fly. At the moment, the server doesn't handle these switches well, and ends up failing.

**Comment by wzJun1** [comment]
> You can pass in the LiveKitOverrides class to solve this problem, for example: `LiveKit.create( appContext = MyApplication.getInstance2(), options = RoomOptions(adaptiveStream = true, dynacast = true), overrides = LiveKitOverrides( okHttpClient = OkHttpClientBuilder.builder(mutableListOf(HttpLoggingInterceptor { LogUtil.d(Constant.TAG_HTTP, it) }.setLevel(BODY))).build(), videoEncoderFactory = HardwareVideoEncoderFactory(EglBase.create().eglBaseContext, false, true), ) )` Resolve using HardwareVideoEncoderFactory.

@MikuMikuQAQ 
@davidliu 

this method can solve the camera problem on Huawei phones or Hongmeng OS, but there is a new problem with screen sharing: other phones cannot decode the screen sharing of Huawei phones or HarmonyOS. 

If you don't use HardwareVideoEncoderFactory, the camera will have problems again, which is really worrying.

**Comment by davidzhao** [comment]
Copying over comment from #245

> I think the better thing to do is to:
> * do not override to `HardwareVideoEncoderFactory`
> * publish videos with VP8 instead of H.264
>   * this can be done either via changing it in your project settings (or config.yaml if self-hosting)
>   * or setting `videoCodec` to vp8 in VideoTrackPublishOptions [here](https://docs.livekit.io/client-sdk-android/livekit-android-sdk/io.livekit.android.room.participant/-local-participant/publish-video-track.html)
